simpleGraph/bytesAlign.go

```diff
@@ -0,0 +1,422 @@
+			//log.Printf("right alignment: %s\n", simpleio.ByteCigarString(rightAlignment))
+		}
+		currScore = leftScore + seedScore + rightScore
+		if currScore > int64(currBest.AlnScore) {
+			currBest.QStart = minQuery
+			currBest.QEnd = maxQuery
+			currBest.PosStrand = currSeed.PosStrand
+
+			currBest.Path = setPath(currBest.Path, minTarget, CatPaths(CatPaths(leftPath, getSeedPath(currSeed)), rightPath), maxTarget)
+			//currBest.Aln = AddSClip(minQuery, len(currSeq), cigar.CatCigar(cigar.AddCigar(leftAlignment, &cigar.Cigar{RunLength: int64(sumLen(currSeed)), Op: 'M'}), rightAlignment))
+
+			currBest.ByteCigar = simpleio.SoftClipBases(minQuery, len(currSeq), simpleio.CatByteCigar(simpleio.AddCigar(leftAlignment, simpleio.ByteCigar{RunLen: sumLen(currSeed), Op: 'M'}), rightAlignment))
+			currBest.AlnScore = int(currScore)
+			currBest.Seq = currSeq
+			if gg.Nodes[currBest.Path.Nodes[0]].Info != nil {
+				currBest.Notes[0].Value = fmt.Sprintf("%s=%d", gg.Nodes[currBest.Path.Nodes[0]].Name, gg.Nodes[currBest.Path.Nodes[0]].Info.Start)
+				currBest.Notes = append(currBest.Notes, infoToNotes(gg.Nodes, currBest.Path.Nodes))
+			} else {
+				currBest.Notes[0].Value = fmt.Sprintf("%s=%d", gg.Nodes[currBest.Path.Nodes[0]].Name, 1)
+			}
+		}
+	}
+	if !currBest.PosStrand {
+		fastq.ReverseQualUint8Record(currBest.Qual)
+	}
+	return &currBest
+}
+
+func RoutineSimplyGiraf(gg *SimpleGraph, seedHash map[uint64][]uint64, seedLen int, stepSize int, scoreMatrix [][]int64, inputChan <-chan *fastq.PairedEndBig, outputChan chan<- *giraf.GirafPair, wg *sync.WaitGroup) {
+	m, trace := simpleio.MatrixSetup(10000)
+	for read := range inputChan {
+		outputChan <- WrapSimplyGsw(gg, read, seedHash, seedLen, stepSize, scoreMatrix, m, trace)
+	}
+	wg.Done()
+}
+
+func WrapSimplyGsw(gg *SimpleGraph, readPair *fastq.PairedEndBig, seedHash map[uint64][]uint64, seedLen int, stepSize int, scoreMatrix [][]int64, m [][]int64, trace [][]simpleio.CigarOp) *giraf.GirafPair {
+	var mappedPair giraf.GirafPair = giraf.GirafPair{Fwd: nil, Rev: nil}
+	mappedPair.Fwd = SimplyGsw(gg, readPair.Fwd, seedHash, seedLen, stepSize, scoreMatrix, m, trace)
+	mappedPair.Rev = SimplyGsw(gg, readPair.Rev, seedHash, seedLen, stepSize, scoreMatrix, m, trace)
+	setGirafFlags(&mappedPair)
+	return &mappedPair
+}
+
+func SoftClipBases(front int, lengthOfRead int, cig []simpleio.ByteCigar) []simpleio.ByteCigar {
+	var runLen int = simpleio.QueryRunLen(cig)
+	if runLen < lengthOfRead {
+		answer := make([]simpleio.ByteCigar, 0, len(cig)+2)
+		if front > 0 {
+			answer = append(answer, simpleio.ByteCigar{RunLen: uint32(front), Op: 'S'})
+		}
+		answer = append(answer, cig...)
+		if front+simpleio.QueryRunLen(cig) < lengthOfRead {
+			answer = append(answer, simpleio.ByteCigar{RunLen: uint32(lengthOfRead - front - runLen), Op: 'S'})
+		}
+		return answer
+	} else {
+		return cig
+	}
+
+}
+
+func SimpleSequenceGraph(ref <-chan *fasta.Fasta, vcfMap map[string][]*vcf.Vcf) *SimpleGraph {
```

Comment:
Maybe this should be in a separate file and a separate pull request from the changes to the alignment code?